tests/ui/append_instead_of_extend.rs

```diff
@@ -0,0 +1,57 @@
+// run-rustfix
+#![warn(clippy::append_instead_of_extend)]
+use std::collections::BinaryHeap;
+fn main() {
+    //gets linted
+    let mut vec1 = vec![0u8; 1024];
+    let mut vec2: std::vec::Vec<u8> = Vec::new();
+
+    vec2.extend(vec1.drain(..));
+
+    let mut vec3 = vec![0u8; 1024];
+    let mut vec4: std::vec::Vec<u8> = Vec::new();
+
+    vec4.extend(vec3.drain(..));
+
+    let mut vec10 = return_vector();
+    let mut vec11: std::vec::Vec<u8> = Vec::new();
+
+    vec11.extend(vec10.drain(..));
```

Comment:
Last thing: I meant to write a test like

```suggestion
    vec11.extend(return_vector().drain(..));
```

without binding it beforehand.